### ISSUE TITLE
Implement basic OnyxDatePicker

### ISSUE THREAD

**Issue by mj-hof** [SchwarzIT/onyx#1818]
### Depends on

- #1817

### Why?

We want to offer a date picker component where users can select a date or datetime, e.g. birthdays or meeting dates.
Because implementing a whole date picker from scratch is time consuming, we want to offer a "quick win" by already
offering a `OnyxDatePicker` component which uses the existing `OnyxInput` and the native HTML `<input type="date" />` for the flyout.
Like this, the component mostly looks and feels like onyx and we can provide the component quicker.

Later, we can refactor it and build our own flyout with more advanced features.

### Design

[Figma](https://www.figma.com/design/YfEUBOHk4J4nYrk04geswG/Onyx-Component-Library?node-id=7420-36065&t=pvuWNaMRTggogXnV-4)

### Acceptance criteria

- a date can be selected
  - [x] manually via keyboard
  - [x] via flyout
- a datetime can be selected
  - [x] manually via keyboard
  - [x] via flyout
- [x] the component considers the current i18n locale and the users browser preference for i18n
- [x] all features of the `OnyxInput` are supported, except properties that do not make sense for dates (e.g. type email)
- [x] if possible, the native HTML flyout aligns with the light/dark mode
- [x] a min and max date can be set
- [x] should work in all timezones

### Implementation details

- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types
- values should be ISO timestamps
- for formatting, we should use the `Intl` API

```ts
export type OnyxDatePickerProps = OnyxInputProps & {
  type?: "date" | "datetime";
};
```

### Definition of Done

- The following component requirements are implemented:
  - [x] skeleton
  - [x] density
  - [x] example usage added to `apps/demo-app/src/views/HomeView.vue`
- covered by tests
  - [x] functional tests (Playwright or unit test)
  - [x] visual tests (Playwright screenshots)
- [x] follow-up tickets were created if necessary
- [x] updated version + documentation is deployed
- [x] Storybook can show the feature
- [x] Storybook code snippet of new/changed examples are checked that they are generated correctly
- [x] Namings are aligned with Figma

**Approval**

- [Storybook](https://storybook.onyx.schwarz/?path=/docs/form-elements-datepicker--docs)
- [x] approved by designer @jannick-ux
- [x] approved by dev @MajaZarkova

**Comment by larsrickert** [comment]
Idea for design: Show current timezone / be able to change timezone

**Comment by larsrickert** [comment]
Idea: Check out new temporal date API

**Comment by jannick-ux** [comment]
The calendar icon should change its color to primary, when hovered/active.
Rest looks fine. Thx
I will approve afterwards

**Comment by larsrickert** [comment]
> The calendar icon should change its color to primary, when hovered/active. Rest looks fine. Thx I will approve afterwards

@jannick-ux The calendar icon is part of the native browser date picker so unfortunately we can not style it for all browser. E.g. it would only be possible to style it for chromium browser (like edge, chrome etc.) but e.g. not for Firefox.

I talked with @JoCa96 and we decided that we want to stick to the native calendar icon for now and when we implement the "real" date picker design in the future, we most likely have to re-implement it anyways so the icon will be updated then.

Is this also fine for you or should we style the icon for those browsers that support it?
